autogpt_platform/frontend/src/components/monitor/FlowRunInfo.tsx

```diff
@@ -1,79 +1,158 @@
+} from "@/lib/autogpt-server-api";
 import { FlowRun } from "@/lib/types";
 import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
 import Link from "next/link";
 import { Button, buttonVariants } from "@/components/ui/button";
 import { IconSquare } from "@/components/ui/icons";
-import { Pencil2Icon } from "@radix-ui/react-icons";
+import { ExitIcon, Pencil2Icon } from "@radix-ui/react-icons";
 import moment from "moment/moment";
 import { FlowRunStatusBadge } from "@/components/monitor/FlowRunStatusBadge";
+import RunnerOutputUI, { BlockOutput } from "../runner-ui/RunnerOutputUI";
 
 export const FlowRunInfo: React.FC<
   React.HTMLAttributes<HTMLDivElement> & {
     flow: GraphMeta;
     flowRun: FlowRun;
   }
 > = ({ flow, flowRun, ...props }) => {
+  const [isOutputOpen, setIsOutputOpen] = useState(false);
+  const [blockOutputs, setBlockOutputs] = useState<BlockOutput[]>([]);
+  const api = useMemo(() => new AutoGPTServerAPI(), []);
+
+  const fetchBlockResults = useCallback(async () => {
+    console.log("fetchBlockResults");
+    const graph = await api.getGraph(flow.id, flow.version);
+    const executionResults = await api.getGraphExecutionInfo(
+      flow.id,
+      flowRun.id,
+    );
+
+    // Filter nodes to only get output blocks
+    const outputNodes = graph.nodes.filter(
+      (node) => node.block_id === "363ae599-353e-4804-937e-b2ee3cef3da4",
+    );
+
+    // Create a map of the latest COMPLETED execution results by node_id
+    const latestCompletedResults = executionResults
+      .filter((result) => result.status === "COMPLETED")
+      .reduce((acc, result) => {
+        const existing = acc.get(result.node_id);
+
+        // Compare dates if there's an existing result
+        if (existing) {
+          const existingDate = existing.end_time || existing.add_time;
+          const currentDate = result.end_time || result.add_time;
+
+          if (currentDate > existingDate) {
+            acc.set(result.node_id, result);
+          }
+        } else {
+          acc.set(result.node_id, result);
+        }
+
+        return acc;
+      }, new Map<string, NodeExecutionResult>());
+
+    // Transform output nodes to BlockOutput format
+    setBlockOutputs(
+      outputNodes.map((node) => ({
+        id: node.id,
+        type: "output" as const,
+        outputSchema: {} as BlockIORootSchema,
+        hardcodedValues: {
+          name:
+            latestCompletedResults.get(node.id)?.input_data.name || "Output",
+          description:
+            latestCompletedResults.get(node.id)?.input_data.description ||
+            "Output from the agent",
+        },
+        result: latestCompletedResults.get(node.id)?.output_data || undefined,
+      })),
+    );
+  }, [api, flow.id, flow.version, flowRun.id]);
+
+  // Fetch graph and execution data
+  useEffect(() => {
+    if (!isOutputOpen || blockOutputs.length > 0) {
+      return;
+    }
+
+    fetchBlockResults();
+  }, [api, isOutputOpen]);
```

Comment:
Does this pass the linting? should `blockOutputs` part of the deps list ? should we ignore the lint and add some comments ?

Reply:
I've fixed deps